iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/recover/WALNodeRecoverTask.java

```diff
@@ -155,7 +155,8 @@ private long[] recoverLastFile() {
         WALEntry walEntry = walReader.next();
         long searchIndex = DEFAULT_SEARCH_INDEX;
         if (walEntry.getType() == WALEntryType.INSERT_TABLET_NODE
-            || walEntry.getType() == WALEntryType.INSERT_ROW_NODE) {
+            || walEntry.getType() == WALEntryType.INSERT_ROW_NODE
+            || walEntry.getType() == WALEntryType.INSERT_ROWS_NODE) {
```

Comment:
There is a bug here, I forgot consider DeleteDataNode here. To fix this, please use WALEntryType.needSearch and consider the search index of DeleteDataNode.

Reply:
fixed